src/query/cellxgene_census/script.py

```diff
@@ -36,128 +37,127 @@ def setup_logger():
+    t1 = adata.shape
     logger.info(
-    "Adding extented metadata to gene expression data."
+        "Removed %s cells based on %s cell_filter_minimum_count of %s cell_filter_grouping."
+        % ((t0[0] - t1[0]), par["cell_filter_minimum_count"], par["cell_filter_grouping"])
     )
-    census_datasets = census_connection["census_info"]["datasets"].read().concat().to_pandas()
-    
-    query_data.obs.dataset_id = query_data.obs.dataset_id.astype("category")
-
-    dataset_info = census_datasets[census_datasets.dataset_id.isin(query_data.obs.dataset_id.cat.categories)]\
-    [['collection_id', 'collection_name', 'collection_doi', 'dataset_id', 'dataset_title']]\
-    .reset_index(drop=True)\
-    .apply(lambda x: x.astype('category'))
-
-    return query_data.obs.merge(
-        dataset_info, on='dataset_id', how = 'left'
+    return adata
+
+def filter_by_counts(adata, par):
+    logger.info("Remove cells with few counts and genes with few counts.")
+    t0 = adata.shape
+    # remove cells with few counts and genes with few counts
+    if par["cell_filter_min_counts"]:
+        sc.pp.filter_cells(adata, min_counts=par["cell_filter_min_counts"])
+    if par["cell_filter_min_genes"]:
+        sc.pp.filter_cells(adata, min_genes=par["cell_filter_min_genes"])
+    if par["gene_filter_min_counts"]:
+        sc.pp.filter_genes(adata, min_counts=par["gene_filter_min_counts"])
+    if par["gene_filter_min_cells"]:
+        sc.pp.filter_genes(adata, min_cells=par["gene_filter_min_cells"])
+    t1 = adata.shape
+    logger.info("Removed %s cells and %s genes.", (t0[0] - t1[0]), (t0[1] - t1[1]))
+
+def move_x_to_layers(adata):
+    logger.info("Move .X to .layers['counts']")
+    adata.layers["counts"] = adata.X
+    adata.X = None
+
+def print_unique(adata, column):
+    formatted = "', '".join(adata.obs[column].unique())
+    logger.info(f"Unique {column}: ['{formatted}']")
+
+def print_summary(adata):
+    logger.info(f"Resulting dataset: {adata}")
+
+    logger.info("Summary of dataset:")
+    print_unique(adata, "assay")
+    print_unique(adata, "assay_ontology_term_id")
+    print_unique(adata, "cell_type")
+    print_unique(adata, "cell_type_ontology_term_id")
+    print_unique(adata, "dataset_id")
+    print_unique(adata, "development_stage")
+    print_unique(adata, "development_stage_ontology_term_id")
+    print_unique(adata, "disease")
+    print_unique(adata, "disease_ontology_term_id")
+    print_unique(adata, "tissue")
+    print_unique(adata, "tissue_ontology_term_id")
+    print_unique(adata, "tissue_general")
+    print_unique(adata, "tissue_general_ontology_term_id")
+
+def write_anndata(adata, par):
+    logger.info("Writing MuData object to '%s'", par["output"])
+
+    mdata = mu.MuData({par["output_modality"]: ad.AnnData()})
+
+    mdata.write_h5mu(par["output"], compression=par["output_compression"])
+
+    mu.write_h5ad(par["output"], data=adata, mod=par["output_modality"])
```

Comment:
I would just write a .h5mu file directly. Just to make sure that compression works.

Reply:
+1